### PR TITLE
Fix CosineSimilarity metric dropping axis in get_config

### PR DESCRIPTION
**Pull request by sohumt123**
## Description

`keras.metrics.CosineSimilarity` accepts an `axis` argument (documented, and used in the class docstring examples), but its `get_config` returned a hard-coded `{"name": ..., "dtype": ...}` and dropped `axis`. After a `serialize`/`deserialize` round trip — e.g. saving and reloading a model compiled with this metric — `from_config(**config)` fell back to the default `axis=-1`, and the restored metric silently computed different numbers.

Repro on master:

```python
from keras import metrics

m = metrics.CosineSimilarity(axis=0)
m2 = metrics.deserialize(metrics.serialize(m))

m.update_state([[0., 1.], [1., 1.]], [[1., 0.], [1., 1.]])
m2.update_state([[0., 1.], [1., 1.]], [[1., 0.], [1., 1.]])
print(float(m.result()))   # 0.7071067690849304
print(float(m2.result()))  # 0.4999999701976776  (axis silently reset to -1)
```

**Root cause:** `CosineSimilarity` subclasses `MeanMetricWrapper` and overrides `get_config` (to avoid serializing the wrapped `fn`), but the override only kept `name`/`dtype` and omitted `axis`.

**Fix:** store `self.axis` in `__init__` and include it in `get_config`, matching the pattern already used by the sibling `MeanMetricWrapper` subclasses `PearsonCorrelation` and `ConcordanceCorrelation` in `correlation_metrics.py`.

**Testing:** extended `CosineSimilarityTest::test_config` to assert `axis` survives `from_config(get_config())`, and added `test_config_preserves_axis_result`, which asserts a restored metric with a non-default `axis` produces the same result as the original. Both new assertions fail on master and pass with this change. Full run: `pytest keras/src/metrics/regression_metrics_test.py keras/src/metrics/reduction_metrics_test.py keras/src/metrics/metric_test.py` → 66 passed (TensorFlow backend).

**AI disclosure (per the AI-Assisted Contribution Policy):** I used a general-purpose AI coding assistant while preparing this change. I have reviewed and tested the code and take full responsibility for it.

## Contributor Agreement

**Please review our [AI-Assisted Contribution Policy](../CONTRIBUTING.md#ai-assisted-contribution-policy) and check all boxes below before submitting your PR for review:**

- [x] I am a human, and not a bot.
- [x] I will be responsible for responding to review comments in a timely manner.
- [x] I will work with the maintainers to push this PR forward until submission.

**Note:** Failing to adhere to this agreement may result in your future PRs no longer being reviewed.

### CODE REVIEW

**Comment by JyotinderSingh** [comment]
Please fix the failing CI test.
